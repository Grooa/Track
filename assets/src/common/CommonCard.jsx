import PropTypes from 'prop-types';

import ReactMarkdown from 'react-markdown';

const CommonCard = (props) => {
  const {
    link = null,
    title = null,
    type = null,
    description = null,
    buttonLabel = null,
    isDisabled = false,
  } = props;

  const typeRendered = type ? <strong className="type">{type}</strong> : '';
  const titleRendered = title ? <h3 className="title">{title}</h3> : '';
  let buttonRendered = buttonLabel ? <button>{buttonLabel}</button> : <button>Checkout</button>;

  if (isDisabled && !buttonLabel) {
    buttonRendered = <button>You don't have acccess</button>;
  }

  const header = <div className="course-module-metadata">
    {typeRendered}
    {titleRendered}
  </div>;

  const body = description
    ? <ReactMarkdown className="description" source={description} />
    : '';

  let cssClasses = 'course-module';
  if (link && !isDisabled) {
    cssClasses += ' clickable';
  } else if (isDisabled) {
    cssClasses += ' inactive';
  }

  if (!link) {
    return <div className={cssClasses}>
      {header}

      {body}

      {buttonRendered}
    </div>;
  }

  return <a href={link} className={cssClasses}>
    {header}

    {body}

    {buttonRendered}
  </a>;
};

CommonCard.propTypes = {
  link: PropTypes.string,
  title: PropTypes.string,
  type: PropTypes.string,
  description: PropTypes.string,
  buttonLabel: PropTypes.string,
  isDisabled: PropTypes.bool,
};

export default CommonCard;
