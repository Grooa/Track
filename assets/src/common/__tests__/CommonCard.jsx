import { render, shallow } from 'enzyme';

import CommonCard from '../CommonCard';

describe('<CommonCard />', () => {
  let title;
  let type;
  let description;
  let buttonLabel;
  let link;
  let isDisabled;

  beforeEach(() => {
    title = 'Hello world';
    type = 'module';
    description = '**Lorem lipsum** dolor sit amed';
    buttonLabel = 'Watch';
    link = 'https://test.com/test';
    isDisabled = false;
  });

  it('should render properly', () => {
    const wrapper = render(<CommonCard
      title={title}
      type={type}
      description={description}
      buttonLabel={buttonLabel}
      link={link}
      isDisabled={isDisabled}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render disabled properly', () => {
    const wrapper = render(<CommonCard
      title={title}
      type={type}
      description={description}
      buttonLabel={buttonLabel}
      isDisabled={true} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render empty block properly', () => {
    const wrapper = render(<CommonCard/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should use <div> as wrapper when link is empty', () => {
    const wrapper = shallow(<CommonCard link={null} />);

    expect(wrapper.find('div').length).toBe(2);
  });

  it('should use <a> as wrapper when link is not empty', () => {
    const wrapper = shallow(<CommonCard link={link}/>);

    expect(wrapper.find('a').length).toBe(1);
  });

  it('should set className `inactive` when isDisabled is true', () => {
    const wrapper = shallow(<CommonCard link={link} isDisabled={true}/>);

    expect(wrapper.find('a').hasClass('inactive')).toBeTruthy();
  });

  it('should not have className `clickable` when isDisabled is true', () => {
    const wrapper = shallow(<CommonCard link={link} isDisabled={true}/>);

    expect(wrapper.find('a').hasClass('clickable')).toBeFalsy();
  });

  it('should set className `clickable` when link is not empty', () => {
    const wrapper = shallow(<CommonCard link={link}/>);

    expect(wrapper.find('a').hasClass('clickable')).toBeTruthy();
  });
});
