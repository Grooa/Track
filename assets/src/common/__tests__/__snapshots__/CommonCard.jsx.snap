// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommonCard /> should render disabled properly 1`] = `
<div
  class="course-module inactive"
>
  <div
    class="course-module-metadata"
  >
    <strong
      class="type"
    >
      module
    </strong>
    <h3
      class="title"
    >
      Hello world
    </h3>
  </div>
  <div
    class="description"
  >
    <p>
      <strong>
        Lorem lipsum
      </strong>
       dolor sit amed
    </p>
  </div>
  <button>
    Watch
  </button>
</div>
`;

exports[`<CommonCard /> should render empty block properly 1`] = `
<div
  class="course-module"
>
  <div
    class="course-module-metadata"
  />
  <button>
    Checkout
  </button>
</div>
`;

exports[`<CommonCard /> should render properly 1`] = `
<a
  class="course-module clickable"
  href="https://test.com/test"
>
  <div
    class="course-module-metadata"
  >
    <strong
      class="type"
    >
      module
    </strong>
    <h3
      class="title"
    >
      Hello world
    </h3>
  </div>
  <div
    class="description"
  >
    <p>
      <strong>
        Lorem lipsum
      </strong>
       dolor sit amed
    </p>
  </div>
  <button>
    Watch
  </button>
</a>
`;
